src/applications/personalization/profile-2/components/MobileMenuTrigger.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState, useRef } from 'react';
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+
+import {
+  closeSideNav as closeSideNavAction,
+  openSideNav as openSideNavAction,
+  pinMenuTrigger as pinMenuTriggerAction,
+  unpinMenuTrigger as unpinMenuTriggerAction,
+} from '../actions';
+
+const MobileMenuTrigger = ({
+  closeSideNav,
+  focusTriggerButton,
+  openSideNav,
+  isMenuTriggerPinned,
+  pinMenuTrigger,
+  unpinMenuTrigger,
+}) => {
+  const button = useRef(null);
+  const placeholder = useRef(null);
+
+  const [mobileBreakpoint] = useState(767);
+  const [isMobile, setIsMobile] = useState(false);
+  const [triggerPosition, setTriggerPosition] = useState(null);
+  const [triggerHeight, setTriggerHeight] = useState(null);
+
+  // init some local state
+  useEffect(
+    () => {
+      setTriggerHeight(window.getComputedStyle(button.current).height);
+      setTriggerPosition(placeholder.current.offsetTop);
+      setIsMobile(window.innerWidth <= mobileBreakpoint);
+    },
+    [mobileBreakpoint],
+  );
+
+  useEffect(() => {
```

Comment:
Why do we declare `useEffect` twice?

Reply:
the code in each one will get triggered under different cases. The first one will only get trigged if the value of the `mobileBreakpoint` changes. Which it never actually will. So the first one effectively functions as a `componentDidMount` replacement. The second one will fire when _any_ props or values in the component change.